### ISSUE TITLE
How do you obtain a unique output for a given input mask?

### ISSUE THREAD

**Issue by jeanRassaire** [mobaidoctor/med-ddpm#22]
Hello,

I'm trying to use this model on my own dataset. I would like to know if I can get a unique output for a given input mask. The sampling process required some initialization with random sampling, i.e. x_T ~N(0, I). This initialization would definetly have a impact on the output. So how do you guarantee a unique output for a given mask?

**Comment by mobaidoctor** [comment]
Hi @jeanRassaire, thank you again for your inquiry and interest in our work. Could you provide more details about what you mean by "unique output"? Since this is a generative model, it produces different brain features with each inference. Each output is unique because our method converts random noise into images. However, if the training dataset is small (fewer than 100 samples), the generated images might appear similar, though not identical. Given the high-dimensional nature of volumetric images,  the probability of generating the exact same image during inference is extremely low.

**Comment by jeanRassaire** [comment]
Hi @mobaidoctor  thank you for your reply. I am  the code for segmentation. What I mean by a unique output is that given an input volume, you get the same output mask at all times, rather than getting different output masks.

**Comment by mobaidoctor** [comment]
@jeanRassaire Oh, for a segmentation task, you'll need to modify the model architecture slightly because segmentation tasks typically output a binary image. This means the input image is used as a condition, and our current method might not reach high accuracy for these tasks. We've tested our method on polyp segmentation, and while it did produce segmentation results, it didn't perform as well as state-of-the-art methods. We recommend changing the model architecture. You should also consider using a hybrid loss function. Additionally, implementing enhancement techniques for your conditional images, especially since they are volumetric, could further improve performance.

**Comment by jeanRassaire** [comment]
Thanks @mobaidoctor

**Comment by HIT-wwb** [comment]
@jeanRassaire I am also interested in using this code to generate pseudo-CT images from MRI, how is your progress

**Comment by rassaire** [comment]
@HIT-wwb, As @mobaidoctor mentioned, you need a large number of epochs for the model to converge. For my project, it's a work in progress because I'm not spending a lot of time on it. I've started training the model and have been trying to get it to generalize, but I'm still having problems with overfitting.
